source/contracts/reporting/Auction.sol

```diff
@@ -0,0 +1,239 @@
+        uint256 _attoRepCost = _attoEthAmount.mul(_ethPriceInAttoRep) / 10**18;
+        reputationToken.trustedAuctionTransfer(msg.sender, this, _attoRepCost);
+        msg.sender.transfer(_attoEthAmount);
+        uint256 _attoEthSold = initialAttoEthBalance.sub(currentAttoEthBalance);
+        uint256 _newTotalAttoEthSold = _attoEthSold.add(_attoEthAmount);
+        uint256 _repPriceInAttoEth = (10**36) / _ethPriceInAttoRep;
+        currentLowerBoundRepPriceInAttoEth = currentLowerBoundRepPriceInAttoEth
+            .mul(_attoEthSold)
+            .add(_attoEthAmount.mul(_repPriceInAttoEth))
+            .div(_newTotalAttoEthSold);
+        currentAttoEthBalance = currentAttoEthBalance.sub(_attoEthAmount);
+        currentRepPrice = currentLowerBoundRepPriceInAttoEth.add(currentUpperBoundRepPriceInAttoEth) / 2;
+        return true;
+    }
+
+    function tradeEthForRep(uint256 _attoRepAmount) public initializeNewAuctionIfNeeded payable returns (bool) {
+        require(currentAttoRepBalance > 0);
+        require(_attoRepAmount > 0);
+        _attoRepAmount = _attoRepAmount.min(currentAttoRepBalance);
+        uint256 _repPriceInAttoEth = getRepSalePriceInAttoEth();
+        uint256 _attoEthCost = _attoRepAmount.mul(_repPriceInAttoEth) / 10**18;
+        // This will raise an exception if insufficient ETH was sent
+        msg.sender.transfer(msg.value.sub(_attoEthCost));
+        reputationToken.transfer(msg.sender, _attoRepAmount);
+        uint256 _attoRepSold = initialAttoRepBalance.sub(currentAttoRepBalance);
+        uint256 _newTotalAttoRepSold = _attoRepSold.add(_attoRepAmount);
+        currentUpperBoundRepPriceInAttoEth = currentUpperBoundRepPriceInAttoEth
+            .mul(_attoRepSold)
+            .add(_attoRepAmount.mul(_repPriceInAttoEth))
+            .div(_newTotalAttoRepSold);
+        currentAttoRepBalance = currentAttoRepBalance.sub(_attoRepAmount);
+        // We don't update the price while in bootstrap mode since its a one sided auction
+        if (!bootstrapMode) {
+            currentRepPrice = currentLowerBoundRepPriceInAttoEth.add(currentUpperBoundRepPriceInAttoEth) / 2;
+        }
+        return true;
+    }
+
+    function getRepSalePriceInAttoEth() public initializeNewAuctionIfNeeded returns (uint256) {
+        uint256 _timePassed = controller.getTimestamp().sub(initializationTime).sub(currentAuctionIndex * 1 days);
+        uint256 _priceDecrease = initialRepSalePrice.mul(_timePassed) / Reporting.getAuctionDuration();
+        return initialRepSalePrice.sub(_priceDecrease);
+    }
+
+    function getEthSalePriceInAttoRep() public initializeNewAuctionIfNeeded returns (uint256) {
+        require(!bootstrapMode);
+        uint256 _timePassed = controller.getTimestamp().sub(initializationTime).sub(currentAuctionIndex * 1 days);
+        uint256 _priceDecrease = initialEthSalePrice.mul(_timePassed) / Reporting.getAuctionDuration();
+        return initialEthSalePrice.sub(_priceDecrease);
```

Comment:
_priceDecrease could go above initialEthSalePrice if timePassed is greater than duration. I don't know what behavior we want there, but I imagine it's common if dust is left and it isn't worth the gas to take, even for free?

Reply:
In that case the auction would be over and the modifier to initialize would fail since there is no active auction.

Also the sub would raise an exception when it resulted in overflow

Reply:
Maybe that's fine, error means disabled